lib/blockstore/blockstore.go

```diff
@@ -32,35 +30,32 @@ func NewTemporarySync() *SyncStore {
 	return &SyncStore{bs: make(MemStore)}
 }
 
-// WrapIDStore wraps the underlying blockstore in an "identity" blockstore.
-func WrapIDStore(bstore blockstore.Blockstore) blockstore.Blockstore {
-	return blockstore.NewIdStore(bstore)
+// WrapIDStore wraps the underlying blockstore in an identity blockstore.
+// The identity blockstore resolves inlined CIDs immediately, without querying
+// the underlying blockstore.
+func WrapIDStore(bstore blockstore.Blockstore) LotusBlockstore {
+	idstore := blockstore.NewIdStore(bstore)
+	if lbs, ok := idstore.(LotusBlockstore); ok {
+		return lbs
+	}
+	panic("expected IdStore to implement LotusBlockstore")
 }
 
-// NewBlockstore creates a new blockstore wrapped by the given datastore.
-func NewBlockstore(dstore ds.Batching) blockstore.Blockstore {
+// NewFromDatastore creates a new blockstore wrapped by the given datastore.
+func NewFromDatastore(dstore ds.Batching) LotusBlockstore {
 	return WrapIDStore(blockstore.NewBlockstore(dstore))
 }
 
-// Alias so other packages don't have to import go-ipfs-blockstore
+// LotusBlockstore is a standard blockstore enhanced with a view operaiton
+// (zero-copy access to values), and potentially with cache management
+// operations, or others, in the future.
+type LotusBlockstore interface {
+	Blockstore
+	Viewer
+}
```

Comment:
Could we drop the alias below, and call this `Blockstore`? Would let us avoid all the renaming everywhere.

Reply:
That's what I did initially, but I reverted because it causes ambiguity in those places that still take a standard Blockstore (e.g. Graphsync and others). We would end up with two `blockstore.Blockstore` interfaces that are different because one embeds the other...

Reply:
We can still alias that blockstore here, just as something like `BasicBlockstore` (to relay the fact that the non-viewer blockstore is not the one that should be used by default).

Reply:
Fixes in 66336b9.